### PR TITLE
docs(RouterNavigator): Create JSDoc documentation

### PR DESCRIPTION
**Pull request by sbaechler**
Adresses #2256.

Updated the JSDocs of the RouterNavigator component.

It's a start. I haven't quite figured out the documentation system yet.

### CODE REVIEW

**Comment by asialgearoid** [comment]
@sbaechler Thank you very much for a documentation PR! I agree that the process of building documentation is a bit confusing. I will try to add something to the `CONTRIBUTING` file to better explain how to do it. Your PR looks good though. I will test it in the next few days.

**Comment by asialgearoid** [comment]
@sbaechler I have updated `CONTRIBUTING.md` with instructions on how to set up and test the documentation, if you would like to test your own changes. Take a look at https://github.com/OnsenUI/OnsenUI/blob/master/.github/CONTRIBUTING.md#documentation

File: bindings/react/src/components/RouterNavigator.jsx [hunk 1]
```diff
@@ -260,36 +260,29 @@ RouterNavigator.propTypes = {
...
    * @defaultValue null
    * @description
-   *  [en] This array contains the initial route of the navigator,
-   *  which will be used to render the initial pages in the
-   *  renderPage method.
+   *  [en] This object must contain two properties:
+   *  `routeStack`: An array of route objects,
+   *  `processStack`: An array of process objects `{ type: push | pop | reset, route: userRoute }` that
+   *  describe the transition from the current state to the next state.
+   *  To avoid a FLOC, update the `processStack` first. In the done callback update the `processStack` and
```
[the remark this answers is not in the document]
Comment (sbaechler) [reply bindings/react/src/components/RouterNavigator.jsx]: Should have been FLOUC. Flash of unstyled content. I can rephrase that. The issue is that the component can be unmounted before the animation has finished which does not look good.

**Comment by sbaechler** [comment]
@asialgearoid I updated the docs a bit.

**Comment by asialgearoid** [comment]
Great! Thanks very much for the PR! :+1:

**Pull request merged**